convert_old_db/new_schema.h

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef NEW_SCHEMA_H
+#define NEW_SCHEMA_H
+
+#include "../../Current/TypeSystem/struct.h"
+#include "../../Current/TypeSystem/enum.h"
+
+#if 0
+// Original schema:
+OrderedDictionary<User, POLICY> users{"users", instance};
+OrderedDictionary<Card, POLICY> cards{"cards", instance};
+LightweightMatrix<AuthKeyTokenPair, POLICY> auth_token{"auth_token", instance};
+LightweightMatrix<AuthKeyUIDPair, POLICY> auth_uid{"auth_uid", instance};
+LightweightMatrix<CardAuthor, POLICY> card_authors{"card_authors", instance};
+LightweightMatrix<Answer, POLICY> answers{"answers", instance};
+LightweightMatrix<Favorite, POLICY> favorites{"favorites", instance};
+LightweightMatrix<Comment, POLICY> comments{"comments", instance};
+LightweightMatrix<CommentLike, POLICY> comment_likes{"comment_likes", instance};
+LightweightMatrix<CardFlagAsInappropriate, POLICY> flagged_cards{"flagged_cards", instance};
+LightweightMatrix<CommentFlagAsInappropriate, POLICY> flagged_comments{"flagged_comments", instance};
+LightweightMatrix<Notification, POLICY> notifications{"notifications", instance};
+#endif
+
+CURRENT_ENUM(UID, uint64_t){INVALID_USER = 0u};
+CURRENT_ENUM(CID, uint64_t){INVALID_CARD = 0u};
+
+CURRENT_STRUCT(User) {
+  CURRENT_FIELD(uid, UID, UID::INVALID_USER);
+  CURRENT_FIELD(level, uint8_t, 0u);           // User level [0, 9].
+  CURRENT_FIELD(score, uint64_t, score = 0u);  // User score.
+
+  UID key() const { return uid; }
+  void set_key(const UID value) { uid = value; }
+
+  CURRENT_DEFAULT_CONSTRUCTOR(User) {}
+
+  using T_BRIEF = User;
```

Comment:
:+1:
76abedc and https://github.com/c5t/Current/commit/e9e6af2102369e131b0da906960657f6636b68a0 (sorry, the latter went into an already open pull request in C5T/Current).